paddle/fluid/operators/pool_mkldnn_op.cc

```diff
@@ -267,35 +302,74 @@ class PoolMKLDNNGradOpKernel : public paddle::framework::OpKernel<T> {
       auto pool_bwd_pd = mkldnn::pooling_backward::primitive_desc(
           pool_bwd_desc, mkldnn_engine, *pool_pd);
 
+      // reorder between user_diff_dst and pool diff_dst if needed
+      diff_dst_memory = std::make_shared<memory>(user_diff_dst_memory);
```

Comment:
We recommend `unique_ptr`, maybe next time you can change all of them.

Reply:
@tensor-tang Ok.